### PR TITLE
#160577614-User can create, update and delete comment(s)

### PR DESCRIPTION
**Pull request by YAHYA-H**
## What does this PR do
Allows users to create, update and delete a comment

## Description of task to be completed.
creates, gets, deletes and updates comment(s)

| Endpoint        | Functionality        
| ------------- |-------------| 
| GET api/articles/slug/comments| View related comment to article |
| POST api/articles/slug/comments| create comment to article |
| PUT api/articles/slug/comments/pk| update commit to article |
| POST api/articles/slug/comments/pk| create comment-thread to article |
| GET api/articles/slug/comments/pk| View comment related to article|
| DELETE api/articles/slug/comments/pk| Delete comment related to article|

## Creating a comment
The following is the sample of a request data
```
{
  "comment": {
    "body": "His name was my name too."
  }
}
```

## Response
```
{
  "comment": {
    "id": 1,
    "createdAt": "2016-02-18T03:22:56.637Z",
    "updatedAt": "2016-02-18T03:22:56.637Z",
    "body": "It takes a Jacobian",
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg",
      "following": false
    }
  }
}
```

## Background context
The `following` field will be updated as soon as the `following` task is completed.

## Disclaimer
None

## Screenshot
screen shot 2018-10-08 at 23 53 43

## Relevant Pivotal Tracker Stories
[#160577612](https://www.pivotaltracker.com/story/show/160577614)

### CODE REVIEW

**Comment by esirK** [comment]
Since all the blocking tasks have been merged, fix all the current issues.

**Comment by esirK** [comment]
Also check out for typos in your **PR**

**Comment by esirK** [comment]
Follow https://softwareengineering.stackexchange.com/questions/351727/working-on-a-branch-with-a-dependence-on-another-branch-that-is-being-reviewed to remove commits you didn't work on.

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -129,3 +143,110 @@ def destroy(self, request, *args, **kwargs):
         super().destroy(self, request, *args, **kwargs)
 
         return Response({'message': 'The article has been deleted.'})
+
+
+class CommentAPIView(ListCreateAPIView):
```
Comment (esirK) [inline authors/apps/articles/views.py:433]: I don't think we need to have two classes to create a `Comment` you can use a `ListModelMixin` to provide the listing functionality to the second class

Comment (YAHYA-H) [reply authors/apps/articles/views.py:433]: 😄 okay, I will work on it.

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -129,3 +143,110 @@ def destroy(self, request, *args, **kwargs):
...
+    def filter_queryset(self, queryset):
+        """This method filter and get comment of an article."""
+        filters = {self.lookup_field: self.kwargs[self.lookup_url_kwarg]}
+        return queryset.filter(**filters)
+
+    def create(self, request, *args, **kwargs):
+        """This methods creates a comment"""
+        try:
+            article = Article.objects.get(slug=kwargs['slug'])
+        except Exception:
```
Comment (esirK) [inline authors/apps/articles/views.py]: What exception are you expecting here??

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -129,3 +143,110 @@ def destroy(self, request, *args, **kwargs):
...
+            return Response({'Error,Please check your url?'},
```
Comment (esirK) [inline authors/apps/articles/views.py]: I don't think you should be handling this by yourself. The `url_pattern` should handle it,

File: authors/apps/articles/views.py [hunk 4]
```diff
@@ -129,3 +143,110 @@ def destroy(self, request, *args, **kwargs):
...
+        serializer.save(article=article, author=request.user.profile)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class CommentCreateUpdateDestroy(CreateAPIView, RetrieveUpdateDestroyAPIView):
+    """This class view creates update and delete comment"""
+    queryset = Comment.objects.all()
+    serializer_class = CommentSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly, )
+    lookup_url_kwarg = "pk"
```
Comment (esirK) [inline authors/apps/articles/views.py:472]: `pk` is the default so you can ignore it.

File: authors/apps/articles/views.py (hunk 1 above)
Comment (YAHYA-H) [reply authors/apps/articles/views.py:433]: please kindly clarify for me, do you mean i should combine the two classes i created for the comment? @esirK

File: authors/apps/articles/views.py (hunk 3 above)
Comment (YAHYA-H) [reply authors/apps/articles/views.py]: @esirK, do you mean the response ?`url_pattern`handles the slugs error but it's not in JSON format

**Comment by esirK** [comment]
Looks good. Just fix the conflicts

**Pull request merged**